sos/report/plugins/openstack_designate.py

```diff
@@ -0,0 +1,95 @@
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, RedHatPlugin
+
+
+class OpenStackDesignate(Plugin):
+
+    short_desc = 'Openstack Designate'
+
+    plugin_name = "openstack_designate"
+    profiles = ('openstack', 'openstack_controller')
+
+    var_puppet_gen = "/var/lib/config-data/puppet-generated/designate"
+
+    def setup(self):
+        # configs
+        self.add_copy_spec([
+            "/etc/designate/*",
+            self.var_puppet_gen + "/etc/designate/designate.conf",
+            self.var_puppet_gen + "/etc/designate/pools.yaml",
+        ])
+
+        # logs
+        if self.get_option("all_logs"):
+            self.add_copy_spec([
+                "/var/log/designate/*",
+                "/var/log/containers/designate/*",
+            ])
+        else:
+            self.add_copy_spec([
+                "/var/log/designate/*.log",
+                "/var/log/containers/designate/*.log"
+            ])
+
+        # commands
+        self.add_cmd_output('openstack dns service list --all-projects',
+                            subdir='service')
+        self.add_cmd_output('openstack dns quota list --all-projects',
+                            subdir='quota')
+        self.add_cmd_output('openstack zone list --all-projects',
+                            subdir='zone')
+        self.add_cmd_output('openstack ptr record list --all-projects',
+                            subdir='ptr')
+        self.add_cmd_output('openstack zone export list --all-projects',
+                            subdir='export')
+        self.add_cmd_output('openstack zone import list --all-projects',
+                            subdir='import')
+        self.add_cmd_output('openstack zone blacklist list --all-projects',
+                            subdir='blacklist')
+        self.add_cmd_output('openstack zone transfer request list '
+                            '--all-projects', subdir='transfer-request')
+        self.add_cmd_output('openstack zone transfer accept list '
+                            '--all-projects', subdir='transfer-accept')
+        self.add_cmd_output('openstack tld list --all-projects',
+                            subdir='tld')
+        self.add_cmd_output('openstack tsigkey list --all-projects '
+                            '--column name --column algorithm --column scope',
+                            subdir='tsig')
```

Comment:
The `subdir` parameter is typically used when there will be numerous, iterative, outputs for a particular item. It looks here like we'll just have a single output in each of these subdirs, so we really don't gain anything here by using them.

If there are individual command outputs to be captured for each item being listed here, then those should be added and contained in the same subdir - otherwise I think it's better if these outputs are in the top-level plugin directory together.